docs/001_develop/02_server-capabilities/018_model_context_protocol/index.mdx

```diff
@@ -0,0 +1,201 @@
+  HostApp -->|Invoke Tool| MCPClient
+  MCPClient <-->|MCP Protocol| MCPServer
+  MCPServer <--> EventHandler
+  MCPServer <--> RequestServer
+```
+
+## What is MCP?
+
+MCP (Model Context Protocol) is a standard interface that exposes application capabilities in a format that LLMs can understand and interact with. These capabilities are described via metadata, which lets an LLM know:
+- What operations are available
+- What inputs are required
+- What responses to expect
+
+Because all Genesis Event Handlers and Request Servers automatically generate and publish metadata describing the request and response formats, any one of these endpoints can easily be enabled for interaction via AI through MCP.
+
+More information on MCP can be found on the official website, [here](https://modelcontextprotocol.io/introduction)
+
+## Getting Started
+
+### Step 1: Enable the MCP Server Process
+
+As an experimental feature, MCP is disabled for applications by default and is an opt-in feature. In order to enable MCP and have the `GENESIS_MCP` process appear in your application, add the following sysdef item to your system-definition.kts file:
+
+```kotlin
+item(name = "GENESIS_MCP_PROCESS_START", value = "true")
+```
+
+### Step 2: Configure your application MCP server
+
+By default, the MCP server opens an HTTP server on port 3001. This can be configured by creating a file called genesis-mcp-server.kts, which will override the default service config.
+Other attributes such as the server name, version and how often the server will scan for new tools can also be configured.
+
+The service configuration offers several options for added security through different authentication strategies. These are described in depth in a later section.
+
+```kotlin
+server {
+    serverName = "genesis" // The name of your MCP server
+    version = "1.0.0" // The name of your MCP server
+    port = 3001 // The port the HTTP server will run on
+    toolRefreshIntervalMillis = 10000L // The amount of time the server will wait before checking the app metadata again for any new resources
+
+    authenticationStrategy {
+        staticUser {
+            userName = "admin"
+            requiresApproval = true
+        }
+    }
+}
+```
+
+### Step 3: Enable Event Handlers and Request Servers for MCP
+
+We then need to enable individual resources for MCP, as we do not want to expose all resources by default. We can do that by creating an `<application>-mcp.kts` file in `<your-application-module>/src/main/genesis/scripts`.
+
+Here is an example file with an example entry:
+
+```kotlin
+mcp {
+    enableMcp("MY_ENDPOINT") {
+        this.context = """
+            Type your context here
+        """.trimIndent()
+    }
+}
+```
+
+The context is provided to the LLM as the description of the MCP tool that will be created to call that resource. This text can be as long as required, and should include specific examples and guidelines on how to use the tool, to guide the LLM.
```

Comment:
This is good but I think I would also find it useful to have a concrete example use case with a "real" endpoint and the kind of context you'd add to it, to help people understand the concept. The document is completely abstract currently and I am not sure it would help the concept click for people who don't already understand it